packages/wells/src/client/api/surveys.ts

```diff
@@ -0,0 +1,62 @@
+import { Sequence, SequencesAPI } from '@cognite/sdk';
+import { SequenceRow } from 'stable/dist/src/api/sequences/sequenceRow';
+import { Survey, SearchSurveys } from '../model/Survey';
+
+export class Surveys extends SequencesAPI {
+  /**
+   * Maps from sequence array to a survey array.
+   * Contains lazy getter for rows
+   *
+   * @param sequences
+   */
+  private mapToSurvey = (sequences: Sequence[]): Survey[] => {
+    return sequences.map(sequence => {
+      return <Survey>{
+        id: sequence.id,
+        name: sequence.name,
+        description: sequence.description,
+        assetId: sequence.assetId,
+        externalId: sequence.externalId,
+        metadata: sequence.metadata,
+        rows: async (limit: number = 100): Promise<SequenceRow[]> => {
```

Comment:
Will this promise be lazy or eager? I.e. when you create a Survey object, will it always fetch all its rows by default, or only when the user tries to access survey.rows?
Since the data can be quite heavy, especially if you have a method that returns several surveys, we probably want this to be lazy, so if the user is only interested in metadata, they don't have to wait for all the data to be loaded

Again, I don't know a lot of JS, but from quickly reading this link - https://tusharsharma.dev/posts/js-promises-eager-not-lazy - it seems you may want to have the rows as a method in the Survey data class instead of an attribute?

Reply:
rows is implemented as a method, could have written `async function rows(limit: number = 100) : Promise<SequenceRow>` to get the same. It is a lazy implementation and I also double-checked it when running the tests, so rows() will not be called when the a survey/trajectory is instantiated, it will only be called when you actually run the rows method: `const rows = await trajectories[someIndex].rows(100)`.

Reply:
same with wellbore.trajectories(), it is also function when the object is instantiated: 
![image](https://user-images.githubusercontent.com/25263842/98250065-4d193900-1f77-11eb-8dc8-be1f93b33052.png)

Reply:
Ok, so the surveys class is the one that actually holds the reference to the client and then you have to have this to call mapToSurvey? So the wellbore has to ive inside this class so it can access the `this` property?